Umm, please better check what each commit has to say 😅🙏

Minecraftian14 opened the pull request.

Minecraftian14 committed:
Update Black Hole and For Addition

The 2 things changes in this commit are
* Well, added BlackHoles 😁
* Changed how forces are added, dunno if that's better but please check it out first 🙏.

Minecraftian14 committed:
Opps, i forgot to remove the old variables....

alyrow committed:
Delete gradle.xml

Since I need to delete .idea

alyrow committed:
Delete misc.xml

Since I need to delete .idea

Minecraftian14 committed:
OMG! Lol

This time i added:
* White holes, push the particles away from a point. 😋
* Black Lines
-> Mathematically saying, BL attaracts particles perpendicularly towrards the line.
-> Generally: It just attracts the particles towards a point.
* White Lines, umm you need an explanation 🤣?

Minecraftian14 committed:
Merge branch 'pre-master' of https://github.com/Minecraftian14/box2d-light-particle into pre-master

alyrow commented on src/main/java/com/alyrow/gdx/particle/physics/BlackHole.java:
```diff
@@ -3,22 +3,27 @@
+        cache.x = center.x - particle.x;
+        cache.y = center.y - particle.y;
+        rs = cache.len2();
+        return cache.nor().scl(effect * particle.mass / rs);
```
When rs < 1, particles gain light speed

Minecraftian14 committed:
Added Revolution

alyrow commented on src/main/java/com/alyrow/gdx/particle/physics/BlackLine.java:
```diff
@@ -0,0 +1,31 @@
+    public Vector2 getForce(PhysicParticle particle) {
+        cache = line.normal(particle.x, particle.y);
+        rs = (float) Math.pow(line.distance(particle.x, particle.y), 2);
+        return cache.scl(-effect * particle.mass / rs);
```
Same bug with rs < 1

Minecraftian14 committed:
OMG, read the desc

This time the forces i added are
* Whirlpool -> Self explanatory i hope?
* PointAttract -> Lineat attraction towards a point.
* Fan -> All particles move with a constant angular velocity about a point.
* Drain -> Similar to whirlpool but attraction towards the center is linear, ie, constant velocity.

Then i also added a CompoundForce class, which just takes in an array of a lot of forces and produces the resultant.

alyrow reviewed (approved):
Nice! Very complete 🥇
